NitroxClient/Communication/Client.cs

```diff
@@ -54,7 +74,14 @@ private void DataReceived(IAsyncResult ar)
                 packetReceiver.PacketReceived(packet);                
             }
 
-            connection.BeginReceive(new AsyncCallback(DataReceived));
+            if (connection.Open)
+            {
+                connection.BeginReceive(new AsyncCallback(DataReceived));
+            } else
```

Comment:
minor: mixing convention (else on new line) throughout these files. I'll add a style checker later to catch these.